### PR TITLE
Exit gracefully when requests encounter a ReadTimeout exception.

### LINKED ISSUE #1923: Show a better error message when there's an HTTP timeout
Using docker 1.6.2 and docker-compose 1.3.3, I occasionally get this error on docker-compose up for one of the containers it tries to spin up.  I think this happens more when I do a docker load on a saved image before but not 100% certain on that.

Here's the callstack.  It's not a blocking issue since it's intermittent but a better error message handling this exception would be nice
```
Traceback (most recent call last):
  File "/usr/local/bin/docker-compose", line 9, in <module>
    load_entry_point('docker-compose==1.3.3', 'console_scripts', 'docker-compose')()
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/main.py", line 32, in main
    command.sys_dispatch()
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/docopt_command.py", line 21, in sys_dispatch
    self.dispatch(sys.argv[1:], None)
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/command.py", line 34, in dispatch
    super(Command, self).dispatch(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/docopt_command.py", line 24, in dispatch
    self.perform_command(*self.parse(argv, global_options))
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/command.py", line 66, in perform_command
    handler(project, command_options)
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/main.py", line 471, in up
    timeout=timeout
  File "/usr/local/lib/python2.7/dist-packages/compose/project.py", line 245, in up
    timeout=timeout
  File "/usr/local/lib/python2.7/dist-packages/compose/service.py", line 318, in execute_convergence_plan
    do_build=do_build,
  File "/usr/local/lib/python2.7/dist-packages/compose/service.py", line 225, in create_container
    return Container.create(self.client, **container_options)
  File "/usr/local/lib/python2.7/dist-packages/compose/container.py", line 38, in create
    response = client.create_container(**options)
  File "/usr/local/lib/python2.7/dist-packages/docker/client.py", line 245, in create_container
    return self.create_container_from_config(config, name)
  File "/usr/local/lib/python2.7/dist-packages/docker/client.py", line 252, in create_container_from_config
    res = self._post_json(u, data=config, params=params)
  File "/usr/local/lib/python2.7/dist-packages/docker/clientbase.py", line 128, in _post_json
    return self._post(url, data=json.dumps(data2), **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/docker/clientbase.py", line 83, in _post
    return self.post(url, **self._set_request_timeout(kwargs))
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 508, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 465, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/adapters.py", line 433, in send
    raise ReadTimeout(e, request=request)
```

### PR DESCRIPTION
**Pull request by shin-**
Fixes #1923

### CODE REVIEW

**Comment by dnephin** [comment]
I think if we're going to add `COMPOSE_HTTP_TIMEOUT` we need to add it to the docs as well.

What's the advantage over using `DOCKER_CLIENT_TIMEOUT` ? Is the idea that users can set them to different values if they need to?

**Comment by aanand** [comment]
It's more that `COMPOSE_HTTP_TIMEOUT` is a better variable name, because in general we shouldn't invent `DOCKER_*` variables - that's the domain of the Docker client.

File: compose/cli/main.py [hunk 1]
```diff
@@ -65,6 +66,12 @@ def main():
     except NeedsBuildError as e:
         log.error("Service '%s' needs to be built, but --no-build was passed." % e.service.name)
         sys.exit(1)
+    except ReadTimeout as e:
+        log.error(
+            "HTTP request took too long to complete. Retry with --verbose to obtain debug information.\n"
+            "If you encounter this issue regularly because of slow network conditions, consider setting "
+            "COMPOSE_HTTP_TIMEOUT to a higher value."
```
Comment (aanand) [inline compose/cli/main.py]: "An HTTP request..." reads better, I think.

Would be good to print the current value of the variable here.

File: compose/cli/docker_client.py [hunk 2]
```diff
@@ -34,5 +34,5 @@ def docker_client():
             ca_cert=ca_cert,
         )
 
-    timeout = int(os.environ.get('DOCKER_CLIENT_TIMEOUT', 60))
+    timeout = int(os.environ.get('COMPOSE_HTTP_TIMEOUT', os.environ.get('DOCKER_CLIENT_TIMEOUT', 60)))
```
Comment (aanand) [inline compose/cli/docker_client.py]: We should deprecate `DOCKER_CLIENT_TIMEOUT`, i.e. show a deprecation warning if it's set, much as we do with the `FIG_*` variables: https://github.com/docker/compose/blob/17682c58db8986f1d1492559d72b3960b86f5f05/compose/cli/command.py#L92-L94

**Comment by shin-** [comment]
How about now? :)

**Comment by aanand** [comment]
Great. There's now a bit of duplication and inconsistency - the default value is hardcoded twice, and we're only falling back to `DOCKER_CLIENT_TIMEOUT` in one place, not both.

Could you move it all into one place?  Perhaps populate a constant in `const.py`.

**Comment by shin-** [comment]
:+1:

**Comment by aanand** [comment]
LGTM

**Pull request merged**